crates/turbopack-ecmascript/src/chunk/optimize.rs

```diff
@@ -267,6 +269,27 @@ async fn merge_duplicated_and_contained(
     Ok(())
 }
 
+async fn merge_by_size(chunks: &[EcmascriptChunkVc]) -> Result<Vec<EcmascriptChunkVc>> {
+    let mut merged = Vec::new();
+    let mut current = Vec::new();
+    let mut current_items = 0;
+    for &chunk in chunks {
+        let chunk_items = *chunk.chunk_items_count().await?;
+        if current_items == 0 || current_items + chunk_items < MAX_CHUNK_ITEMS_PER_CHUNK {
+            current.push(chunk);
+            current_items += chunk_items;
+        } else {
+            merged.push(merge_chunks(*current.first().unwrap(), &current).await?);
```

Comment:
This could panic if the the chunk is extremely large and it's the first item in the `chunks` slice.
```suggestion
            merged.push(merge_chunks(*current.first().unwrap_or(current), &current).await?);
```